### ISSUE TITLE
Problem on Linux with exif_call()

### ISSUE THREAD

**Issue by insilentio** [JoshOBrien/exiftoolr#4]
Hi Josh
Sorry, it's me again. I am currently trying to move some picture resizing code to RStudio Workbench, which runs under Linux. I have encountered the following problem. While install_exiftool() and configure_exiftoolr() seem to work properly, exif_call() etc. do not work.

e.g.
_>` exif_version()`
sh: 'perl' '/home/sszbad/R/x86_64-pc-linux-gnu-library/4.0/exiftoolr/exiftool/exiftool': No such file or directory
Error in system2(exiftoolpath, args = args, stdout = stdout) : 
  error in running command_

I tried some debugging and am now somehow closer to the reason, albeit not the solution of the problem.

The test_exiftool function in configure_exiftoolr() uses system(), not system2() to run the exiftool command against the OS:
_Browse[2]> `test_exiftool`
```
function (command, quiet = TRUE) 
{
    if (!quiet) 
        message("Trying exiftool command: ", command)
    command_works <- suppressWarnings(suppressMessages(0 == try(**system**(command, 
        ignore.stdout = TRUE, ignore.stderr = TRUE, show.output.on.console = TRUE), 
        silent = TRUE)))
    if (command_works) {
        ver_string <- paste(system(command, ignore.stderr = TRUE, 
            intern = TRUE), collapse = "\n")
        ver_number <- suppressWarnings(as.numeric(ver_string))
        return(!is.na(ver_number))
    }
    else {
        return(FALSE)
    }
}
```
<bytecode: 0x55b476576da0>
<environment: namespace:exiftoolr>_

And this call works properly:
_Browse[3]> `system(command, ignore.stdout = TRUE, ignore.stderr = TRUE, show.output.on.console = TRUE)`
arguments 'show.output.on.console', 'minimized' and 'invisible' are for Windows only
Browse[3]> `command`
[1] "'perl' '/home/sszbad/R/x86_64-pc-linux-gnu-library/4.0/exiftoolr/exiftool/exiftool' -ver"
Browse[3]>` system(command)`
12.34_

However, exif_call() etc. seem to use system2(). Now, if I understand the differrence between system and system2 correctly, system2 does not allow args in the command call ("Unlike system, command is always quoted by shQuote, so it must be a single command without arguments.")

I believe this is the reason why it fails. The exif_call() tries to run this command:
_`system2(exiftoolpath, args = args, stdout = stdout)`_
where in my case
_Browse[3]> `exiftoolpath`
[1] "'perl' '/home/sszbad/R/x86_64-pc-linux-gnu-library/4.0/exiftoolr/exiftool/exiftool'"_
, while the same command but with system() instead of system2() works.

If I manually change the exiftoolpath above to just "perl" and add the real exiftool path to args, then the call works:
_Browse[3]>   `system2(exiftoolpath, args = args, stdout = stdout)`
[1] "12.34"
Browse[3]> `exiftoolpath`
[1] "perl"
Browse[3]> `args`
[1] "/home/sszbad/R/x86_64-pc-linux-gnu-library/4.0/exiftoolr/exiftool/exiftool"
[2] "-@"                                                                        
[3] "'/tmp/RtmpI3pgul/args.cmd152c152d23795'"_

I assume that the system call under Linux is behaving differently than under Win or MacOS (where I didn't found such issues).

Kind regards and sorry for the really lengthy description
Daniel

**Comment by JoshOBrien** [comment]
@insilentio Good catch, and thanks for getting started with the debugging. I think, with cf8715918, I've implemented a fix. (The commit message for 38dabad describes the underlying issue that was causing the problems you reported.)

Would you mind installing the most recent version (using `devtools::install_github("JoshOBrien/exiftoolr")` or otherwise), and testing on your Linux and then letting me know how it goes? If it works for you, I'll test a bit more and then push a new version to CRAN.

**Comment by insilentio** [comment]
@JoshOBrien will try to do it tomorrow

**Comment by insilentio** [comment]
@JoshOBrien, installed the latest version and this is working fine, thanks a lot.

**Comment by JoshOBrien** [comment]
@insilentio Super. Thanks.

**Comment by JoshOBrien** [comment]
Closed with 38dabad290ad380ce294ad5e98375631974caf00 and cf871591872792d147d961a76cf64c7d4b4e8fba.
